src/requestApproveTemplate.ts

```diff
@@ -0,0 +1,53 @@
+import { S3 } from 'aws-sdk';
+import { Md5 } from 'ts-md5/dist/md5';
+import * as fs from 'fs';
+import * as cli from 'cli-color';
+import * as path from 'path';
+
+import { Arguments } from 'yargs';
+import { loadStackArgs, parseS3HttpUrl } from './cfn/index';
+import configureAWS from './configureAWS';
+import {logger} from './logger';
+
+export async function requestApproveTemplate(argv: Arguments): Promise<number> {
+  const stackArgs = await loadStackArgs(argv);
+  const s3 = new S3();
+
+  await configureAWS(stackArgs.Profile, stackArgs.Region);
+
+  const cfnTemplate = await fs.readFileSync(stackArgs.Template);
+  const s3Url = parseS3HttpUrl(stackArgs.ApprovedTemplateLocation);
```

Comment:
`ApprovedTemplateLocation` isn't a HTTP URL, is it? It seems like you should just use the `url` library (as the extra logic in `parseS3HttpUrl` isn't going to work on an `s3://` URL):

```
import * as url from 'url';
s3Url = url.parse(stackArgs.ApprovedTemplateLocation);
s3Url.hostname; // bucket
s3Url.path; // key
```